x-pack/plugins/security/server/routes/views/access_agreement.ts

```diff
@@ -47,12 +47,22 @@ export function defineAccessAgreementRoutes({
       // authenticated with the help of HTTP authentication), that means we should safely check if
       // we have it and can get a corresponding configuration.
       const sessionValue = await getSession().get(request);
-      const accessAgreement =
-        (sessionValue &&
+
+      let accessAgreement: string = '';
+
+      if (sessionValue) {
+        const localAccessAgreement =
```

Comment:
__optional nit:__ we used "global vs provider-specific" terminology in the past, but I'm happy to change that if you think "local" makes more sense.
```ts
const accessAgreementMessage = providerSpecificAccessAgreement?.message ??
 globalAccessAgreement?.message ?? '';
```

Reply:
done!